test/unit/runner.spec.js

```diff
@@ -496,6 +496,62 @@ describe('Runner', function() {
         runner.failHook(hook, err);
       });
     });
+
+    describe('hugeStackTrace', function() {
+      beforeEach(function() {
+        if (path.sep !== '/') {
+          this.skip();
+        }
+      });
+
+      it('should not hang if the error message is ridiculously long single line', function(done) {
+        var hook = new Hook();
+        hook.parent = suite;
+        var data = [];
+        // mock a long message
+        for (var i = 0; i < 10000; i++) data[i] = {a: 1};
+        var message = JSON.stringify(data);
+        var err = new Error();
+        // Fake stack-trace
+        err.stack = [message].concat(stack).join('\n');
+
+        runner.on('fail', function(hook, err) {
+          expect(
+            err.stack
+              .split('\n')
+              .slice(1)
+              .join('\n'),
+            'to be',
+            stack.slice(0, 3).join('\n')
+          );
+          done();
+        });
+        runner.failHook(hook, err);
+      });
+
+      it('should not hang if error message is ridiculously long multiple lines either', function(done) {
+        var hook = new Hook();
+        hook.parent = suite;
+        var fpath = require('path').join(__dirname, '../../mocha.js');
+        var message = require('fs').readFileSync(fpath, 'utf8');
```

Comment:
I've failed to mock any long but concise text to re-produce regex block. Using `mocha.js` is the easiest I can come up with. @plroebuck

Reply:
updated both `fs` and `path`